Microsoft.Toolkit.Uwp.UI.Controls/Carousel/Carousel.cs

```diff
@@ -0,0 +1,387 @@
+        public double ItemRotationX
+        {
+            get { return (double)GetValue(ItemRotationXProperty); }
+            set { SetValue(ItemRotationXProperty, value); }
+        }
+
+        // Using a DependencyProperty as the backing store for Rotation.  This enables animation, styling, binding, etc...
+        public static readonly DependencyProperty ItemRotationXProperty = DependencyProperty.Register("ItemRotationX", typeof(double), typeof(Carousel), new PropertyMetadata(0d, OnCarouselPropertyChanged));
+
+        /// <summary>
+        /// Gets or sets rotation angle on Y
+        /// </summary>
+        public double ItemRotationY
+        {
+            get { return (double)GetValue(ItemRotationYProperty); }
+            set { SetValue(ItemRotationYProperty, value); }
+        }
+
+        // Using a DependencyProperty as the backing store for Rotation.  This enables animation, styling, binding, etc...
+        public static readonly DependencyProperty ItemRotationYProperty = DependencyProperty.Register("ItemRotationY", typeof(double), typeof(Carousel), new PropertyMetadata(0d, OnCarouselPropertyChanged));
+
+        /// <summary>
+        /// Gets or sets rotation angle on Z
+        /// </summary>
+        public double ItemRotationZ
+        {
+            get { return (double)GetValue(ItemRotationZProperty); }
+            set { SetValue(ItemRotationZProperty, value); }
+        }
+
+        // Using a DependencyProperty as the backing store for Rotation.  This enables animation, styling, binding, etc...
+        public static readonly DependencyProperty ItemRotationZProperty = DependencyProperty.Register("ItemRotationZ", typeof(double), typeof(Carousel), new PropertyMetadata(0d, OnCarouselPropertyChanged));
+
+        /// <summary>
+        /// Gets or sets the Carousel orientation. Horizontal or Vertical
+        /// </summary>
+        public Orientation Orientation
+        {
+            get { return (Orientation)GetValue(OrientationProperty); }
+            set { SetValue(OrientationProperty, value); }
+        }
+
+        // Using a DependencyProperty as the backing store for MaxViewableItems.  This enables animation, styling, binding, etc...
+        public static readonly DependencyProperty OrientationProperty = DependencyProperty.Register("Orientation", typeof(Orientation), typeof(Carousel), new PropertyMetadata(Orientation.Horizontal, OnCarouselPropertyChanged));
+
+        /// <summary>
+        /// Each time a property change, we have to reposition
+        /// </summary>
+        private static void OnCarouselPropertyChanged(DependencyObject d, DependencyPropertyChangedEventArgs e)
+        {
+            if (e.NewValue == null)
+            {
+                return;
+            }
+
+            Carousel carouselControl = (Carousel)d;
+            if (carouselControl.ItemsPanel == null)
+            {
+                return;
+            }
+
+            var itemsPanel = carouselControl.GetItemsPanel();
```

Comment:
The panel should be retrieved once during OnApplyTemplate rather than getting it numerous times